controlplane/kubeadm/internal/kubeadm_config_map.go

```diff
@@ -134,6 +139,33 @@ func (k *kubeadmConfig) UpdateEtcdMeta(imageRepository, imageTag string) (bool,
 	return changed, nil
 }
 
+// UpdateCoreDNSImageInfo changes the dns.ImageTag and dns.ImageRepository
+// found in the kubeadm config map
+func (k *kubeadmConfig) UpdateCoreDNSImageInfo(repository, tag string) error {
+	data, ok := k.ConfigMap.Data[clusterConfigurationKey]
+	if !ok {
+		return errors.Errorf("could not find key %q in kubeadm config", clusterConfigurationKey)
+	}
+	configuration, err := yamlToUnstructured([]byte(data))
+	if err != nil {
+		return errors.Wrap(err, "unable to convert YAML to unstructured")
+	}
+	dnsMap := map[string]string{
+		dnsTypeKey:            string(kubeadmv1.CoreDNS),
+		dnsImageRepositoryKey: repository,
+		dnsImageTagKey:        tag,
+	}
+	if err := unstructured.SetNestedStringMap(configuration.UnstructuredContent(), dnsMap, dnsKey); err != nil {
```

Comment:
Do you think it would be better and/or safer to match how `UpdateEtcdMeta` does it, where it only updates imageRepository and imageTag if they differ?

Reply:
Added a check in the `UpdateCoreDNS` to only proceed if the from/to images differ